aws/data_source_aws_network_interface.go

```diff
@@ -0,0 +1,174 @@
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ip_owner_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"public_dns_name": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"public_ip": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"attachment": &schema.Schema{
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"attachment_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"device_index": &schema.Schema{
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"instance_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"instance_owner_id": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"security_groups": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"interface_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ipv6_addresses": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"mac_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"private_dns_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"private_ip": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"private_ips": &schema.Schema{
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"requester_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tagsSchemaComputed(),
+		},
+	}
+}
+
+func dataSourceAwsNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	input := &ec2.DescribeNetworkInterfacesInput{
+		NetworkInterfaceIds: []*string{aws.String(d.Get("id").(string))},
+	}
+
+	resp, err := conn.DescribeNetworkInterfaces(input)
+	if err != nil {
+		return err
+	}
+	if resp == nil || len(resp.NetworkInterfaces) == 0 {
+		return fmt.Errorf("no matching network interface found")
+	}
+	if len(resp.NetworkInterfaces) > 1 {
+		return fmt.Errorf("multiple network interfaces matched")
```

Comment:
Do you mind improving the error message here to also include the ID we looked for? e.g. `multiple network interfaces matched %s", d.Id()`